test/src/main/java/io/strimzi/test/k8s/KubeClient.java

```diff
@@ -604,8 +602,19 @@ public List<ServiceAccount> listServiceAccounts(String namespaceName) {
         return client.serviceAccounts().inNamespace(namespaceName).list().getItems();
     }
 
-    public void createOrReplaceServiceAccount(ServiceAccount serviceAccount) {
-        client.serviceAccounts().inNamespace(serviceAccount.getMetadata().getNamespace()).resource(serviceAccount).createOrReplace();
+    /**
+     * Method for creating the specified ServiceAccount.
+     * In case that the ServiceAccount is already created, it is being updated.
+     * This can be caused by not cleared ServiceAccounts from other tests on in case we shut down the test before the cleanup
+     * phase. It should not have an impact on the functionality, we just update the ServiceAccount.
+     * @param serviceAccount ServiceAccount that we want to create or update
+     */
+    public void createOrUpdateServiceAccount(ServiceAccount serviceAccount) {
+        try {
+            client.serviceAccounts().inNamespace(serviceAccount.getMetadata().getNamespace()).resource(serviceAccount).create();
+        } catch (KubernetesClientException e) {
+            client.serviceAccounts().inNamespace(serviceAccount.getMetadata().getNamespace()).resource(serviceAccount).update();
```

Comment:
You should check for the right error here. What is the reasons for SA not being cleaned up?

Reply:
We are applying the resources from `packaging/install/cluster-operator` as they are, so we are not changing their names. That is a problem in cases when we are installing multiple operators.

I wanted to create an issue for fixing that, because I think that this is out of the scope of this PR (changing the bundle installation and verify it is working for all cases), that's why I added those try-catches into the code.

And about the error, I think that this is correct. Or you mean that I should check that it's really the `Already exists` error?

Reply:
> We are applying the resources from packaging/install/cluster-operator as they are, so we are not changing their names. That is a problem in cases when we are installing multiple operators.
> 
> I wanted to create an issue for fixing that, because I think that this is out of the scope of this PR (changing the bundle installation and verify it is working for all cases), that's why I added those try-catches into the code.

I'm not entirely sure why just because they are from install means they cannot be deleted. But you should add it to the comment then. Separate Pr for fixing that is fine.

> And about the error, I think that this is correct. Or you mean that I should check that it's really the Already exists error?

You should check the cause, yes. Not blindly retry after seeing a very generic exception.

Reply:
> I'm not entirely sure why just because they are from install means they cannot be deleted. But you should add it to the comment then. Separate Pr for fixing that is fine.

Yeah they are normally deleted when the test(s) are finished with the clean up. But when you stop the test in middle of the execution, the clean up is not done, thus the resources stay there. When you want to run another test, it fails just because the resources already exist. So there are like two issues.

> You should check the cause, yes. Not blindly retry after seeing a very generic exception.

Will do.

Reply:
> Yeah they are normally deleted when the test(s) is(are) finished with the clean up. But when you stop the test in middle of the execution, the clean up is not done, thus the resources stay there. When you want to run another test, it fails just because the resources already exist. So there are like two issues.

Fine, so that should be in the comments I guess.

Reply:
> Fine, so that should be in the comments I guess.

I have there this `on in case we shut down the test before the cleanup phase` -> besides the typo, I thought that it will be sufficient. Maybe adding one or two more sentences would make it clearer.